hikari/api/rest.py

```diff
@@ -1154,27 +1458,215 @@ async def execute_webhook(
+            invalid image URLs in embeds; users in `user_mentions` not being
+            mentioned in the message content; roles in `role_mentions` not
+            being mentioned in the message content.
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.NotFoundError
+            If the channel is not found.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """  # noqa: E501 - Line too long
 
     @abc.abstractmethod
     async def fetch_gateway_url(self) -> str:
-        ...
+        """Fetch the gateway url.
+
+        !!! note
+            This endpoint does not require any valid authorization.
+
+        Raises
+        ------
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
 
     @abc.abstractmethod
     async def fetch_gateway_bot(self) -> sessions.GatewayBot:
-        ...
+        """Fetch the gateway sharding requirements for the bot user.
+
+        !!! note
+            The bot user is determined by the token used in the request.
+            Providing a non-`Bot` token will result in undefined behaviour.
+
+        Returns
+        -------
+        hikari.sessions.GatewayBot
+            The gateway bot.
+
+        Raises
+        ------
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
 
     @abc.abstractmethod
     async def fetch_invite(self, invite: invites.Inviteish) -> invites.Invite:
-        ...
+        """Fetch an existing invite.
+
+        Parameters
+        ----------
+        invite : hikari.invites.Inviteish
+            The invite to fetch.
+
+        Returns
+        -------
+        hikari.invites.Invite
+            The requested invite.
+
+        Raises
+        ------
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.NotFoundError
+            If the invite is not found.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
 
     @abc.abstractmethod
     async def delete_invite(self, invite: invites.Inviteish) -> None:
-        ...
+        """Delete an existing invite.
+
+        Parameters
+        ----------
+        invite : hikari.invites.Inviteish
+            The invite to delete.
+
+        Raises
+        ------
+        hikari.errors.Forbidden
+            If you are missing the `MANAGE_GUILD` permission in the guild
+            the invite is from or if you are missing the `MANAGE_CHANNELS`
+            permission in the channel the invite is from.
```

Comment:
Sry if this is just me being sleep deprived but could phrasing here (partially the use of two full sentences in this or statement) not be interpreted as meaning that you need both permissions rather than just one? (this would probably also apply to the other webhook endpoints)